File: app/src/main/java/com/example/marco/weather/Weather/WeatherAdapter.java
package com.example.marco.weather.Weather;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

class WeatherAdapter extends FragmentStatePagerAdapter {
    WeatherViewModel viewModel;

    WeatherAdapter(FragmentManager fm, WeatherViewModel viewModel) {
        super(fm);
        this.viewModel = viewModel;
    }

    @Override
    public int getCount() {
        return viewModel.getLocationsNumber();
    }

    @Override
    public Fragment getItem(int position) {
        return CityWeatherFragment.newInstance(position);
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return viewModel.getLocation(position).getName();
    }
}

File: app/src/main/java/com/example/marco/weather/Weather/WeatherAdapter.kt
package com.example.marco.weather.Weather

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter

internal class WeatherAdapter(fm: FragmentManager, var viewModel: WeatherViewModel) : FragmentStatePagerAdapter(fm) {

    override fun getCount(): Int {
        return viewModel.locationsNumber
    }

    override fun getItem(position: Int): Fragment {
        return CityWeatherFragment.newInstance(position)
    }

    override fun getPageTitle(position: Int): CharSequence {
        return viewModel.getLocation(position).name
    }
}
